api/src/test/java/org/openmrs/obs/AbstractHandlerTest.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.obs;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.openmrs.Obs;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.obs.handler.AbstractHandler;
+import org.openmrs.util.OpenmrsUtil;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ AbstractHandler.class, OpenmrsUtil.class, Context.class })
```

Comment:
Is it possible to write the test without PowerMockRunner? I know that our extensive use of static methods sometimes makes this necessary but we should try to avoid this as much as possible. one annoying reason is that tests using PowerMockRunner do not show up in our coverage which is a known problem https://issues.openmrs.org/browse/TRUNK-5150 ; not sure if there is a solution yet. the reason number one is that we bury dependencies in our code using Context.getXXXService, which makes it hard to test/exchange different implementations.

Reply:
@teleivo I had the same concerns when I wrote the tests and already invested some time to try and find a way to make it work without PowerMock. I didn't find a good solution, but I'm not saying there isn't one. I'm open for suggestions! :P

Reply:
make it a BaseContextSensitiveTest thats the only way it has the Context setup and will add to our coverage.

Reply:
@FloBue you're right that this expands the scope of this ticket quite a lot. Could you create an ad-hoc ticket for this? (And skip this effort for now.)

When it will come to it I can help you setting up Spring Test to test complex obs handlers. I have done something along those lines in Attachments already (see [here](https://github.com/openmrs/openmrs-module-attachments/blob/75b6dc2bdf4ff723a6a2cc9417e25801affe516e/api/src/test/java/org/openmrs/module/attachments/obs/ImageAttachmentHandlerContextSensitiveTest.java#L1) for example.)

Reply:
@mks-d The ad-hoc ticket can be found at https://issues.openmrs.org/browse/TRUNK-5392 . I wasn't sure whether this is more an "Enhancement" or a "Bug".